src/core/mustache/jq_template.rs

```diff
@@ -0,0 +1,184 @@
+use std::sync::Arc;
+
+use nom::branch::alt;
+use nom::bytes::complete::{tag, take_until};
+use nom::combinator::map;
+use nom::multi::many0;
+use nom::sequence::delimited;
+use nom::{Finish, IResult};
+
+use super::{JqRuntimeError, JqTransform, Mustache, PathJqValueString};
+use crate::core::mustache::Segment;
+
+#[derive(Debug, Clone, PartialEq, Hash)]
+/// Used to represent a mixture of getters mustache, jq transformations and
+/// const values templates
+pub struct JqTemplate(pub Vec<JqTemplateIR>);
+
+#[derive(Debug, Clone, PartialEq, Hash)]
+/// The IR for each part of the template
+pub enum JqTemplateIR {
+    JqTransform(JqTransform),
+    Literal(String),
+    Mustache(Mustache),
```

Comment:
JQ needs to be wrapped under mustache not the other way round.

This is valid mustache and valid JQ:
```
{{.a.b.c}}
```
When evaluated the above expression should produce the value on field `c`.

Similarly this is valid mustache and JQ:

```
{{.a.b.c | .d[].e}}
```
When evaluated it produces the a list of values stored in `e`.


This is also valid mustache/JQ however it evaluates to a `String` always

```
Foo {{.a.b.c | .d[].e}} Bar
```

This happens because the output of the JQ template is coerced into a string type because its surrounded with text on the left and right.